oidc/config.go

```diff
@@ -114,6 +119,60 @@ func NewConfig(issuer string, clientID string, clientSecret ClientSecret, suppor
 	return c, nil
 }
 
+// Hash will produce a hash value for the Config, which is suitable to use for
+// comparing two configurations for equality.
+func (c *Config) Hash() (uint64, error) {
+	var h uint64
+	var err error
+	args := []string{}
+	args = append(args, c.Issuer, c.ClientID, string(c.ClientSecret), c.ProviderCA)
+	for _, a := range c.SupportedSigningAlgs {
+		args = append(args, string(a))
+	}
+	args = append(args, c.Scopes...)
+	args = append(args, c.Audiences...)
+	args = append(args, c.AllowedRedirectURLs...)
```

Comment:
Should the slices be sorted before hashing?

Reply:
Yes.  Ty!